akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.http.scaladsl.server.directives
+
+import java.io.File
+
+import akka.http.scaladsl.server.{ MissingFormFieldRejection, Directive1 }
+import akka.http.scaladsl.model.{ ContentType, Multipart }
+import akka.util.ByteString
+import scala.concurrent.Future
+import scala.util.{ Failure, Success, Try }
+import akka.stream.scaladsl._
+
+trait FileUploadDirectives {
+
+  import BasicDirectives._
+  import RouteDirectives._
+  import FutureDirectives._
+  import MarshallingDirectives._
+
+  /**
+   * Streams the bytes of the file submitted using multipart with the given file name into a temporary file on disk.
+   * If there is an error writing to disk the request will be rejected, if there is no such field the request will
+   * be rejected, if there are multiple file parts with the same name, the first one will be used and the subsequent
+   * ones ignored.
+   */
+  def uploadedFile(fieldName: String): Directive1[(FileInfo, File)] =
```

Comment:
these should also work with symbols, to be consistent with the rest of HTTP.
I'll provide a sample how to do that